irohad/ordering/impl/kick_out_proposal_creation_strategy.cpp

```diff
@@ -17,10 +17,9 @@ void KickOutProposalCreationStrategy::onCollaborationOutcome(
     RoundType round, size_t peers_in_round) {
   std::lock_guard<std::mutex> guard(mutex_);
   peers_in_round_ = peers_in_round;
-  while (not requested_count_.empty()
-         and requested_count_.begin()->first <= round) {
-    requested_count_.erase(requested_count_.begin());
-  }
+
+  auto it = requested_count_.upper_bound(round);
+  requested_count_.erase(requested_count_.begin(), it);
```

Comment:
```suggestion
  if (it != requested_count_.end())) {
    requested_count_.erase(requested_count_.begin(), it);
  }
```
Correct me if I am wrong :thinking: 
https://en.cppreference.com/w/cpp/container/map/upper_bound
> If no such element is found, past-the-end (see end()) iterator is returned.

Reply:
If all elements of the container are less or equal, than upper_bound will return 'end'. It means all the data from container will be removed.
aka requested_count_.erase(requested_count_.begin(), requested_count_.end());